C++: Avoid uninitialized variables

Full clang static analyzer output:
```
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:163:19: warning: The left operand of '&' is a garbage value [clang-analyzer-core.UndefinedBinaryOperatorResult]
        if (delta & 1)
                  ^
/workspace/build/android/cmake/debug/x86_64/../../../../../sdk/modules/routing-olp/src/cpp/OlpResponseParsers.cpp:222:5: note: Calling 'polyline_decode<(lambda at /workspace/build/android/cmake/debug/x86_64/../../../../../sdk/modules/routing-olp/src/cpp/OlpResponseParsers.cpp:223:31)>'
    hf::polyline_decode(
    ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:367:9: note: Loop condition is true.  Entering loop body
        while (1)
        ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:369:19: note: Calling 'Decoder::decode_one'
            res = dec.decode_one(lat, lng, z);
                  ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:324:13: note: Assuming field 'm_index' is not equal to field 'm_length'
        if (m_index == m_length)
            ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:324:9: note: Taking false branch
        if (m_index == m_length)
        ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:326:9: note: Taking false branch
        if (!m_lat_conv.decode_value(m_encoded, m_index, m_length, lat))
        ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:328:14: note: Calling 'Converter::decode_value'
        if (!m_lng_conv.decode_value(m_encoded, m_index, m_length, lng))
             ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:158:9: note: 'delta' declared without an initial value
        int64_t delta;
        ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:159:14: note: Calling 'Converter::decode_unsigned_varint'
        if (!decode_unsigned_varint(encoded, index, length, delta))
             ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:126:16: note: Assuming 'index' is >= 'length'
        while (index < length)
               ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:126:9: note: Loop condition is false. Execution continues on line 148
        while (index < length)
        ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:148:13: note: 'shift' is <= 0
        if (shift > 0)
            ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:148:9: note: Taking false branch
        if (shift > 0)
        ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:152:9: note: Returning without writing to 'result'
        return true;
        ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:159:14: note: Returning from 'Converter::decode_unsigned_varint'
        if (!decode_unsigned_varint(encoded, index, length, delta))
             ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:159:9: note: Taking false branch
        if (!decode_unsigned_varint(encoded, index, length, delta))
        ^
../../../../../sdk/modules/routing-olp/include/routing-olp/flexible_polyline.hpp:163:19: note: The left operand of '&' is a garbage value
        if (delta & 1)
                  ^
```

--- a/cpp/include/hf/flexpolyline.h
+++ b/cpp/include/hf/flexpolyline.h
@@ -99,7 +99,7 @@ namespace encoder {
             bool inline static decode_unsigned_varint(const std::string& encoded, uint32_t& index, uint32_t length, int64_t& result) {
                 int64_t delta = 0;
                 short shift = 0;
-                int8_t value;
+                int8_t value = 0;
 
                 while (index < length) {
                     value = decode_char(encoded[index]);
@@ -125,7 +125,7 @@ namespace encoder {
             }
 
             bool decode_value(const std::string& encoded, uint32_t& index, uint32_t length, double& result) {
-                int64_t delta;
+                int64_t delta = 0;
                 if (!decode_unsigned_varint(encoded, index, length, delta)) {
                     return false;
                 }
@@ -232,7 +232,7 @@ namespace encoder {
             }
 
             void inline static decode_header_from_string(const std::string& encoded, uint32_t& index, uint32_t length, uint16_t& header) {
-                int64_t value;
+                int64_t value = 0;
                 /* Decode the header version */
                 if(!Converter::decode_unsigned_varint(encoded, index, length, value)) throw std::invalid_argument("Invalid encoding");
                 if (value != hf::FORMAT_VERSION) throw std::invalid_argument("Invalid format version");
@@ -243,7 +243,7 @@ namespace encoder {
             }
 
             void decode_header() {
-                uint16_t header;
+                uint16_t header = 0;
                 decode_header_from_string(m_encoded, m_index, m_length, header);
                 m_precision = header & 15;  // we pick the first 3 bits only
                 header >>= 4;

--- a/cpp/test/test.cpp
+++ b/cpp/test/test.cpp
@@ -88,6 +88,7 @@ void test_decode1() {
     auto res = hf::polyline_decode("BFoz5xJ67i1B1B7PzIhaxL7Y", [&polyline](double lat, double lng, double z) {
         polyline.push_back({lat, lng});
     });
+    assert_true(res);
     std::vector<std::pair<double, double>> expected{{
         {50.10228, 8.69821},
         {50.10201, 8.69567},
@@ -99,7 +100,7 @@ void test_decode1() {
     for (size_t i = 0; i < polyline.size(); ++i) {
         double delta_lat = std::abs(polyline[i].first - expected[i].first);
         double delta_lng = std::abs(polyline[i].second - expected[i].second);
-        assert_true(delta_lng <= 0.000001);
+        assert_true(delta_lat <= 0.000001);
         assert_true(delta_lng <= 0.000001);
     }
 }
@@ -109,6 +110,7 @@ void test_decode2() {
     auto res = hf::polyline_decode("BF05xgKuy2xCx9B7vUl0OhnR54EqSzpEl-HxjD3pBiGnyGi2CvwFsgD3nD4vB6e", [&polyline](double lat, double lng, double z) {
         polyline.push_back({lat, lng});
     });
+    assert_true(res);
     std::vector<std::pair<double, double>> expected{{
         {52.51994, 13.38663},
         {52.51009, 13.28169},
@@ -126,7 +128,7 @@ void test_decode2() {
     for (size_t i = 0; i < polyline.size(); ++i) {
         double delta_lat = std::abs(polyline[i].first - expected[i].first);
         double delta_lng = std::abs(polyline[i].second - expected[i].second);
-        assert_true(delta_lng <= 0.000001);
+        assert_true(delta_lat <= 0.000001);
         assert_true(delta_lng <= 0.000001);
     }
 }
@@ -136,6 +138,7 @@ void test_decode3() {
     auto res = hf::polyline_decode("BlBoz5xJ67i1BU1B7PUzIhaUxL7YU", [&polyline](double lat, double lng, double z) {
         polyline.push_back({lat, lng, z});
     });
+    assert_true(res);
     std::vector<std::tuple<double, double, double>> expected{{
         {50.10228, 8.69821, 10},
         {50.10201, 8.69567, 20},
@@ -148,7 +151,7 @@ void test_decode3() {
         double delta_lat = std::abs(std::get<0>(polyline[i]) - std::get<0>(expected[i]));
         double delta_lng = std::abs(std::get<1>(polyline[i]) - std::get<1>(expected[i]));
         double delta_z = std::abs(std::get<2>(polyline[i]) - std::get<2>(expected[i]));
-        assert_true(delta_lng <= 0.000001);
+        assert_true(delta_lat <= 0.000001);
         assert_true(delta_lng <= 0.000001);
         assert_true(delta_z <= 0.000001);
     }

